src/main/java/seedu/address/model/pair/exceptions/DuplicatePairException.java

```diff
@@ -0,0 +1,11 @@
+package seedu.address.model.pair.exceptions;
+
+/**
+ * Signals that the operation will result in duplicate Pairs (Pairs are considered duplicates if they have the same
+ * identity).
+ */
+public class DuplicatePairException extends RuntimeException {
+    public DuplicatePairException() {
+        super("Operation would result in duplicate pairs");
+    }
+}
```

Comment:
can you abstract this to `DuplicateEntityException` that takes in the `errMsg` and `Class<?>`? the same logic for checking duplicates seems to be duplicated (no puns intended 😄) at various parts of the codebase. 

what do you think of adding a public static method `handleException(boolean condition)` to replace it?

Reply:
Using DuplicateEntityException<T> will mean will will need a way to retrieve the plural word of each noun, like "pairs" and "persons", will keep it in view